src/stormpath.social-login.js

```diff
@@ -0,0 +1,187 @@
+        var service;
+
+        try {
+          service = $injector.get('$' + providerName + 'Login');
+        } catch (err) {
+          return;
+        }
+
+        service.clientId = provider.clientId;
+        providers[providerName].service = service;
+      });
+    }
+
+    return {
+      /**
+       * @ngdoc function
+       *
+       * @name  stormpath.socialLoginService.$socialLogin#getProviders
+       *
+       * @methodOf stormpath.socialLoginService.$socialLogin
+       *
+       * @returns {promise}
+       *
+       * A promise that is resolved with the list of all available OAuth providers.
+       *
+       * @description
+       *
+       * Returns a list of all OAuth providers, provided by the `/oauth/providers` endpoint.
+       */
+      getProviders: function getProviders() {
+        if (providersPromise) {
+          return providersPromise.promise;
+        }
+
+        providersPromise = $q.defer();
+
+        $http.get(STORMPATH_CONFIG.getUrl('OAUTH_PROVIDERS_ENDPOINT')).then(function(response) {
+          var providers = response.data;
+
+          initProviders(providers);
+
+          providersPromise.resolve(providers);
+        }).catch(providersPromise.reject);
+
+        return providersPromise.promise;
+      }
+    };
+  }
+
+  angular.module('stormpath.socialLogin', ['stormpath.CONFIG'])
+
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$socialLoginProvider
+   *
+   * @description
+   *
+   * Provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+   *
+   * Currently, this provider does not have any configuration methods.
+   */
+  .config(['$injector', 'STORMPATH_CONFIG', function $socialLoginProvider($injector, STORMPATH_CONFIG) {
+    /**
+     * @ngdoc object
+     *
+     * @name stormpath.socialLoginService.$socialLogin
+     *
+     * @description
+     *
+     * The social login service provides methods for letting users logging in with Facebook, Google, etc.
+     */
+    var socialLoginFactory = ['$http', '$q', '$injector', function socialLoginFactory($http, $q, $injector) {
+      return new SocialLoginService(STORMPATH_CONFIG, $injector, $http, $q);
+    }];
+
+    $injector.get('$provide').factory(STORMPATH_CONFIG.SOCIAL_LOGIN_SERVICE_NAME, socialLoginFactory);
+  }])
+
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$spJsLoader
+   *
+   * @description
+   *
+   * The `$spJsLoader` provides a method for loading scripts during runtime.
+   * Used by the social provider services to load their SDKs.
+   */
+  .factory('$spJsLoader', ['$q', function($q) {
+    return {
+      load: function load(id, src, innerHTML) {
+        var deferred = $q.defer();
+        var firstJsElement = document.getElementsByTagName('script')[0];
+        var jsElement = document.createElement('script');
+
+        if (document.getElementById(id)) {
+          deferred.resolve();
+          return deferred.promise;
```

Comment:
I'd remove the `return deferred.promise;` and instead wrap the trailing lines 122-127 in an else.